### ISSUE TITLE
Buildmasters offline ...

### ISSUE THREAD

**Issue by Begasus** [haikuports/haikuports#9799]
Hello, posting here also, it looks like both the buildmasters are down/offline at the moment.
@mmlr can you have a look into reviving them?

```
2023-11-28 17:52:21,465: trying to connect to builder mmlr_x86_64
2023-11-28 17:52:31,477: failed to connect to builder: timed out
2023-11-28 17:52:37,684: connection failed: timed out
```

**Comment by waddlesplash** [comment]
They're back.

**Comment by korli** [comment]
They're back.

**Comment by Begasus** [comment]
64bit down again

**Comment by Begasus** [comment]
Seems to be up again.

**Comment by Begasus** [comment]
Seems 32bit is offline now

**Comment by humdingerb** [comment]
64bit - "all builder lost"
 https://build.haiku-os.org/buildmaster/master/x86_64/?buildrunDir=9957&viewMode=expanded

**Comment by humdingerb** [comment]
And up again. Quick. Thanks!

**Comment by Begasus** [comment]
And 64bit down again for a while ...

**Comment by korli** [comment]
in the log "connection failed: [Errno 28] No space left on device"

**Comment by waddlesplash** [comment]
The packages partition on the server was out of space. I deleted all the `.obsolete` packages, there's now 76G free, so things should work again.

**Comment by korli** [comment]
> The packages partition on the server was out of space. I deleted all the `.obsolete` packages, there's now 76G free, so things should work again.

Both masters look stuck at Feb 12 2024 16:59:09.

**Comment by Begasus** [comment]
Just did a small merge, "at the moment" buildmasters are active again.

**Comment by Begasus** [comment]
And there down again (both) for a few days now.

**Comment by korli** [comment]
Up

**Comment by Begasus** [comment]
Buildmasters are both stuck on scipy since yesterday.

**Comment by waddlesplash** [comment]
Back now.

**Comment by korli** [comment]
Seems the builders are stucked on  "moving package ... to cache..."

**Comment by mmlr** [comment]
> Seems the builders are stucked on "moving package ... to cache..."

This is a known issue and caused by Haiku and the builder VMs being to fast. Underlying is a Paramiko issue that fails to notice the remote process is done when it conpletes too quickly. This instance is already fixed in https://github.com/haikuports/haikuporter/commit/8d0c7b054930ab6d3a19d11bcf6babb24b5f5ec7 but that is not yet deployed.

A temporary workaround has been applied in the containers manually a couple of times, but it gets reverted when they are recreated. So at this point this is an infrastructure deployment issue and not on the Haiku side. The fix could also be independently applied without bringing in the rest of the changes related to object storage.

**Comment by Begasus** [comment]
32bit stuck atm with: Cleaning work directory of devilutionx_x86-1.5.3 ...

**Comment by korli** [comment]
> A temporary workaround has been applied in the containers manually a couple of times, but it gets reverted when they are recreated. So at this point this is an infrastructure deployment issue and not on the Haiku side. The fix could also be independently applied without bringing in the rest of the changes related to object storage.

I take creating a branch on the current 1.2.8-6 commit, applying the fix, then bumping to 1.2.8-7 would be possible for me. But building/publishing the docker image would still need the admin team. @kallisti5 what do you think?

**Comment by waddlesplash** [comment]
Re-applied the fix on the buildmasters, they're back again.

**Comment by Begasus** [comment]
Off-line again :(

**Comment by korli** [comment]
Off-line again :(

**Comment by waddlesplash** [comment]
They look fine here, what's the problem?

**Comment by korli** [comment]
x86_64 stucked on Cleaning work directory of kopeninghours24-24.08.3 ...
x86_gcc2 stucked on Cleaning work directory of quazip1_qt6_x86-1.4 ...

**Comment by waddlesplash** [comment]
Containers on the server had been restarted and lost the patch. Applied it again and restarted the workers. (@kallisti5, when are we switching to the S3-based HaikuPorter? :)

**Comment by jmairboeck** [comment]
The builders are down again since yesterday when I rebase-merged #11394 (to keep the recipes in separate commits). Does the buildmaster not like it when pushing multiple commits at once?

**Comment by mmlr** [comment]
> The builders are down again since yesterday when I rebase-merged #11394 (to keep the recipes in separate commits). Does the buildmaster not like it when pushing multiple commits at once?

No, that has nothing to do with it. The builders are firewalled and only allow the buildmaster to connect, as otherwise the flood of connections coming from all the bruteforce attempts bog down our network stack. But since the infrastructure is deployed in a cluster where the source address can change whenever the containers are recreated, the firewall rule gets out of date. Also the hotfix for the issue above is then generally lost and needs to be reapplied (until a new container image is used where the fix is permanent).

This is usually the cause when both builders go down at the same time and needs some manual work. Generally the stability has been better than when exposing the network stack to the open internet, that's why it is done this way right now. As I do not have access to the infrastructure, I depend on someone who has to report the new source address to me, usually waddlesplash. We fix these behind the scenes, but sometimes it just takes a few days, especially during these seasons.

**Comment by Begasus** [comment]
@mmlr @waddlesplash buildmasters are both down since yesterday again:

```
2025-02-16 18:29:04,018: portstree head is at 1f138cf2996b5f911dcb1a64564b376664945243
2025-02-16 18:29:17,150: scheduling build of telegram_desktop_x86-5.11.1
2025-02-16 18:29:17,153: starting build 85444, telegram_desktop_x86-5.11.1 on builder mmlr_x86_gcc2
2025-02-16 18:29:33,372: build 85444, telegram_desktop_x86-5.11.1 failed
2025-02-16 18:29:33,372: transient error, rescheduling build
2025-02-16 18:29:33,372: builder mmlr_x86_gcc2 is reconnecting
```

**Comment by Begasus** [comment]
Could someone look at 32bit buildmaster, it's stuck in "waiting for package .... to be activated" for a few days already?

@mmlr @waddlesplash @kallisti5

**Comment by waddlesplash** [comment]
Rebooted, seems better now.

**Comment by Begasus** [comment]
Thanks! If there no comment on the PR for rav1e and friends I'll start pushing them in the morning.

**Comment by Begasus** [comment]
It's time again :) 32bit buildmaster lost for a few days now:
```
2026-04-12 15:01:30,085: connection failed: [Errno None] Unable to connect to port 15490 on 65.21.32.54
2026-04-12 15:01:30,086: marking builder lost after 10 consecutive transient failures
```

**Comment by korli** [comment]
32-bit x86 builder MIA since 2026-06-28 14:35:05,540

@mmlr @waddlesplash @kallisti5

**Comment by Begasus** [comment]
Something is up at the buildmasters, the packages for Qt5 are stuck on 32bit, attempting a trigger shows the system isn't unmounted cleanly.
https://build.haiku-os.org/buildmaster/master/x86_gcc2/logviewer.html?buildruns/9901/builds/89207.log
Builds for CudaText and qt5_script have finished on 64bit buildmaster but they don't show up when trying to update these packages with pkgman, even after a few attempts with "pkgman refresh".
@mmlr @waddlesplash @kallisti5

**Comment by Begasus** [comment]
Packages build on 64bit still don't show up, like qt5_script update, nothing shows at the depot server:
https://depot.haiku-os.org/#!/?bcguid=bc15-XYGG&repos=haikuports&arch=x86_64&incldev=false&onlynatv=false&onlydstp=false&viewcrttyp=FEATURED&srchexpr=qt5_script

Qt5 build finished on 32bit and that one also doesn't show in the depot server:
https://depot.haiku-os.org/#!/pkg/qt5_x86/haikuports/haikuports_x86_gcc2/5/15/16/-/2/x86_gcc2?bcguid=bc406-BANL

**Comment by Begasus** [comment]
Build packages from buildmasters are still not available in the depot nor can be updated/installed with pkgman, without this being functional it's no use in pushing updates to haikuports. Could someone please have a look what's going on here?

**Comment by lonemadmax** [comment]
The new (and old) packages are available for direct download. It seems it's the part that updates the repo and obsoletes old versions that is failing.

**Comment by Begasus** [comment]
Where are they then? qt5_script isn't showing at the depot server:

https://depot.haiku-os.org/#!/?bcguid=bc15-XYGG&repos=haikuports&arch=x86_64&incldev=false&onlynatv=false&onlydstp=false&viewcrttyp=FEATURED&srchexpr=qt5_script

**Comment by lonemadmax** [comment]
Direct download: https://eu.hpkg.haiku-os.org/haikuports/master/x86_64/current/packages/qt5_script-5.15.19-2-x86_64.hpkg

The depot will show what the repo has, and the repo doesn't seem to have been updated.

**Comment by waddlesplash** [comment]
Should be fixed now.

**Comment by Begasus** [comment]
I already did a local build, so it doesn't hurt me atm, but the users are still stuck on not having the package(s) show up. :)

**Comment by Begasus** [comment]
Buildmastes are stuck again? @mmlr @waddlesplash @kallisti5
